DigitalLearningSolutions.Data/Services/UserService.cs

```diff
@@ -73,5 +76,34 @@ public List<CentreUserDetails> GetUserCentres(AdminUser? adminUser, List<Delegat
 
             return availableCentres.OrderByDescending(ac => ac.IsAdmin).ThenBy(ac => ac.CentreName).ToList();
         }
+
+        public bool TryUpdateUserAccountDetails(int? adminId, int? delegateId, string password, string firstName, string surname, string email)
+        {
+            var (loggedInAdminUser, loggedInDelegateUser) = GetUsersById(adminId, delegateId);
+
+            var signedInEmail = loggedInAdminUser?.EmailAddress ?? loggedInDelegateUser?.EmailAddress;
+
+            var (adminUser, delegateUsers) = GetUsersByEmailAddress(signedInEmail);
+
+            var (verifiedAdminUser, verifiedDelegateUsers) =
+                loginService.VerifyUsers(password, adminUser, delegateUsers);
```

Comment:
Could we pull this into its own private method - get all linked accounts or something similar? Also, I think we should handle this by returning no users:

> Possible null reference argument for parameter 'emailAddress' in '(AdminUser?, List<DelegateUser>) UserService.GetUsersByEmailAddress(string emailAddress)'.

Reply:
so new method GetAllAccounts(int? adminId, int? delegateId, string password) returning the output of the loginService call?

Reply:
Yep, the new private method here which is effectively encapsulating these lines:

```
var (loggedInAdminUser, loggedInDelegateUser) = GetUsersById(adminId, delegateId);

            var signedInEmail = loggedInAdminUser?.EmailAddress ?? loggedInDelegateUser?.EmailAddress; // or return no users

            var (adminUser, delegateUsers) = GetUsersByEmailAddress(signedInEmail);

            return loginService.VerifyUsers(password, adminUser, delegateUsers);
```